hotshot-types/src/traits/election.rs

```diff
@@ -31,56 +29,64 @@ pub trait Membership<TYPES: NodeType>: Debug + Send + Sync {
     fn stake_table(
         &self,
         epoch: Option<TYPES::Epoch>,
-    ) -> Vec<<TYPES::SignatureKey as SignatureKey>::StakeTableEntry>;
+    ) -> Result<Vec<PeerConfig<TYPES::SignatureKey>>, Self::Error>;
```

Comment:
Is it necessary to wrap these returns in `Result`? All the impls seem infallible and the uses simply do something like `unwrap_or_default()`.

Reply:
This was part of tech-debt that we discussed on zulip few weeks ago. Basically, we wanted this to return Option or Result. I agree that this doesn't do much because of  unwrap_or_default() but that requires more work.